libs/common/src/platform/abstractions/crypto.service.ts

```diff
@@ -345,4 +375,38 @@ export abstract class CryptoService {
     encBuffer: EncArrayBuffer,
     key: SymmetricCryptoKey,
   ): Promise<Uint8Array>;
+
+  /**
+   * Retrieves all the keys needed for decrypting Ciphers
+   * @param userId The user id of the keys to retrieve or null if the user is not Unlocked
+   * @param legacySupport `true` if you need to support retrieving the legacy version of the users key, `false` if
+   * you do not need legacy support. Use `true` by necessity only. Defaults to `false`. Legacy support is for users
+   * that may not have updated to use the new {@link UserKey} yet.
+   *
+   * @throws If an invalid user id is passed in.
+   */
+  abstract cipherDecryptionKeys$(
+    userId: UserId,
+    legacySupport?: boolean,
+  ): Observable<CipherDecryptionKeys | null>;
+
+  /**
+   * Gets an observable of org keys for the given user.
+   * @param userId The user id of the user of which to get the keys for.
+   * @return An observable stream of the users organization keys if they are unlocked, or null if the user is not unlocked.
+   * The observable will stay alive through locks/unlocks.
+   *
+   * @throws If an invalid user id is passed in.
+   */
+  abstract orgKeys$(userId: UserId): Observable<Record<OrganizationId, OrgKey> | null>;
+
+  /**
+   * Gets an observable stream of the users public key. If the user is does not have
+   * a {@link UserKey} or {@link UserPrivateKey} that is decryptable, this will emit null.
+   *
+   * @param userId The user id of the user of which to get the public key for.
+   *
+   * @throws If an invalid user id is passed in.
+   */
+  abstract userPublicKey$(userId: UserId): Observable<UserPublicKey>;
```

Comment:
Same question here about throwing if we're in a state that the user doesn't have a decryptable keypair but does have a user key.

```suggestion
  abstract userPublicKey$(userId: UserId): Observable<UserPublicKey | null>;
```